src/testGuess.js

```diff
@@ -0,0 +1,10 @@
+function testGuess(userInput, goal) {
+    if(userInput === '') throw 'error: userInput is empty';
```

Comment:
Noticed your comment about having issues with the testing for invalid numbers - because you're using a number input in the html, you're always going to get a number input (NaN won't be an issue, and it will automatically error in the console if the user doesn't do any input so you won't get a ''). If you want the user to be able to put in things other than numbers, you can just use a normal text input, but I think for the case of this app you should keep it at a number input. Instead, you should approach "invalid numbers" as numbers that are out of range (above 20 or below   1). I'm going to give you the extra credit point on this though because your syntax for throwing errors was correct - just note that this will throw in the console and not actually display for the user so you'll want to add a message for that.